From a8425e69449b97ff452e1883e321e9807054a1ed Mon Sep 17 00:00:00 2001
From: Ed Beroset <beroset@ieee.org>
Date: Sat, 21 Jan 2023 13:51:04 -0500
Subject: [PATCH] Iterate over const ref in importToNotebook()

The previous code had this line for iteration:

  for (const KCalendarCore::Incidence::Ptr &incidence : mEventList) {

The gcc compiler rightly complains that the loop variable 'incidence' of type 'const Ptr&'
{aka 'const QSharedPointer<KCalendarCore::Incidence>&'} binds to a temporary constructed from
type 'const QSharedPointer<KCalendarCore::Event>'

This version fixes that problem:

  for (const QSharedPointer<KCalendarCore::Event>& incidence : mEventList) {

Signed-off-by: Ed Beroset <beroset@ieee.org>
---
 src/calendarimportmodel.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calendarimportmodel.cpp b/src/calendarimportmodel.cpp
index aadce78..f941da2 100644
--- a/src/calendarimportmodel.cpp
+++ b/src/calendarimportmodel.cpp
@@ -158,7 +158,7 @@ QVariant CalendarImportModel::data(const QModelIndex &index, int role) const
 
 bool CalendarImportModel::importToNotebook(const QString &notebookUid, bool discardInvitation) const
 {
-    for (const KCalendarCore::Incidence::Ptr &incidence : mEventList) {
+    for (const QSharedPointer<KCalendarCore::Event>& incidence : mEventList) {
         const KCalendarCore::Incidence::Ptr old =
             mStorage->calendar()->incidence(incidence->uid(), incidence->recurrenceId());
         if (old) {
